src/agl/WebRuntimeAGL.cpp

```diff
@@ -306,91 +331,92 @@ void WebAppLauncherRuntime::setup_signals() {
-      if (tokenv) {
-        m_token = tokenv;
-        m_url.push_back(url_match_result[7].length() ? '&' : '?');
-        m_url.append("token=");
-        m_url.append(m_token);
-      }
-    }
+	if (std::regex_match(m_url, url_match_result, url_regex)) {
+		unsigned counter = 0;
+		for (const auto& res : url_match_result) {
+			LOG_DEBUG("    %d: %s", counter++, res.str().c_str());
+		}
 
-    std::string path = std::string(getenv("AFM_APP_INSTALL_DIR"));
-    if (path.empty()) {
-	    LOG_DEBUG("Please set AFM_APP_INSTALL_DIR");
-	    return false;
-    }
-    path = path + "/" + WEBAPP_CONFIG;
+		if (url_match_result[4].length()) {
+			std::string authority = url_match_result[4].str();
+			std::size_t n = authority.find(':');
+			if (n != std::string::npos) {
+				std::string sport = authority.substr(n+1);
+				m_host = authority.substr(0, n);
+				m_port = stringTo<int>(sport);
+			} else {
+				m_host = authority;
+			}
+		}
 
-    // Parse config file of runxdg
-    if (parse_config(path.c_str())) {
-      LOG_DEBUG("Error in config");
-      return false;
-    }
+		bool url_misses_token = true;
+		if (url_match_result[7].length()) {
+			std::string query = url_match_result[7].str();
+			std::size_t n = query.find('=');
+			if (n != std::string::npos) {
+				m_token = query.substr(n+1);
+				url_misses_token = false;
+			}
+		}
 
-    // Special cases for windowmanager roles
-    if (m_id.rfind("webapps-html5-homescreen", 0) == 0)
-      m_role = "homescreen";
-    else if (m_id.rfind("webapps-homescreen", 0) == 0)
-      m_role = "homescreen";
+		if (url_misses_token) {
+			char *tokenv = getenv("CYNAGOAUTH_TOKEN");
+			if (tokenv) {
+				m_token = tokenv;
+				m_url.push_back(url_match_result[7].length() ? '&' : '?');
+				m_url.append("token=");
+				m_url.append(m_token);
+			}
+		}
 
-    LOG_DEBUG("id=[%s], name=[%s], role=[%s], url=[%s], host=[%s], port=%d, token=[%s], width=[%s], height[%s], surface_type[%d], panel_type[%d]",
-            m_id.c_str(), m_name.c_str(), m_role.c_str(), m_url.c_str(),
-            m_host.c_str(), m_port, m_token.c_str(), m_width.c_str(), m_height.c_str(), m_surface_type, m_panel_type);
+		std::string path = std::string(getenv("AFM_APP_INSTALL_DIR"));
+		if (path.empty()) {
+			LOG_DEBUG("Please set AFM_APP_INSTALL_DIR");
+			return false;
+		}
+		path = path + "/" + WEBAPP_CONFIG;
 
-    // Setup HomeScreen API
-    if (!init_hs()) {
-      LOG_DEBUG("cannot setup hs API");
-      return false;
-    }
+		// Parse config file of runxdg
+		if (parse_config(path.c_str())) {
+			LOG_DEBUG("Error in config");
+			return false;
+		}
 
-    return true;
-  } else {
-    LOG_DEBUG("Malformed url.");
-    return false;
-  }
+		// Special cases for windowmanager roles
+		if (m_id.rfind("webapps-html5-homescreen", 0) == 0)
+			m_role = "homescreen";
+		else if (m_id.rfind("webapps-homescreen", 0) == 0)
+			m_role = "homescreen";
+		else {
+			m_role = m_id.substr(0,12);
```

Comment:
Just fyi, these are from a patch, in the chromium84 tree in AGL project, which shouldn't be actually here. I'll have to remove these.